scripts/check-docker-pin.sh

```diff
@@ -0,0 +1,14 @@
+# This script is checks to that all our Docker images are pinned to a specific SHA256 hash
+#
+# References as to why...
+#   - https://nickjanetakis.com/blog/docker-tip-18-please-pin-your-docker-image-versions
+#   - https://snyk.io/blog/10-docker-image-security-best-practices/ (Specifically: USE FIXED TAGS FOR IMMUTABILITY)
+#
+find ~+ -type f -name "Docker*" -print0 | xargs -0 grep -s "FROM" {} | grep -v scratch | grep -v sha256 | grep -v "solana AS builder" | grep -v "solana AS ci_tests" | grep -v "node_modules"
+if [ $? -eq 0 ]; then
+   echo "[!] Unpinned docker files"
```

Comment:
errors go to stderr:

```suggestion
   echo "[!] Unpinned docker files" >&2
```